altinn-client/src/main/java/no/difi/meldingsutveksling/AltinnWsClient.java

```diff
@@ -96,56 +121,46 @@ private LogstashMarker markerFrom(ReceiptExternalStreamedBE receiptAltinn) {
+        }
+    }
 
+    private BrokerServiceSearch getBrokerServiceSearch(String orgnr) {
         BrokerServiceSearch searchParameters = new BrokerServiceSearch();
         searchParameters.setFileStatus(BrokerServiceAvailableFileStatus.UPLOADED);
         searchParameters.setReportee(orgnr);
         ObjectFactory of = new ObjectFactory();
         JAXBElement<String> serviceCode = of.createBrokerServiceAvailableFileExternalServiceCode(configuration.getExternalServiceCode());
         searchParameters.setExternalServiceCode(serviceCode);
         searchParameters.setExternalServiceEditionCode(configuration.getExternalServiceEditionCode());
-
-        BrokerServiceAvailableFileList filesBasic;
-        try {
-            filesBasic = service.getAvailableFilesBasic(configuration.getUsername(), configuration.getPassword(), searchParameters);
-        } catch (IBrokerServiceExternalBasicGetAvailableFilesBasicAltinnFaultFaultFaultMessage e) {
-            log.error(AVAILABLE_FILES_ERROR_MESSAGE, AltinnReasonFactory.from(e));
-            return Lists.newArrayList();
-        }
-
-        return filesBasic.getBrokerServiceAvailableFile()
-            .stream().map(f -> new FileReference(f.getFileReference(), f.getReceiptID()))
-            .collect(Collectors.toList());
+        return searchParameters;
     }
 
-
-    public EduDocument download(DownloadRequest request) {
-        BrokerServiceExternalBasicStreamedSF brokerServiceExternalBasicStreamedSF;
-        brokerServiceExternalBasicStreamedSF = new BrokerServiceExternalBasicStreamedSF(configuration.getStreamingServiceUrl());
-        IBrokerServiceExternalBasicStreamed streamingService = brokerServiceExternalBasicStreamedSF.getBasicHttpBindingIBrokerServiceExternalBasicStreamed(new MTOMFeature(true));
-        BindingProvider bp = (BindingProvider) streamingService;
-        bp.getRequestContext().put(BindingProvider.ENDPOINT_ADDRESS_PROPERTY, configuration.getStreamingServiceUrl().toString());
-        bp.getRequestContext().put(JAXWSProperties.REQUEST_TIMEOUT, REQUEST_TIMEOUT);
-        bp.getRequestContext().put(JAXWSProperties.CONNECT_TIMEOUT, CONNECT_TIMEOUT);
-        SOAPBinding binding = (SOAPBinding) bp.getBinding();
-        binding.setMTOMEnabled(true);
-
-        byte[] message;
+    public StandardBusinessDocument download(DownloadRequest request, MessagePersister messagePersister) {
         try {
-            message = streamingService.downloadFileStreamedBasic(configuration.getUsername(), configuration.getPassword(), request.getFileReference(), request.getReciever());
-            AltinnPackage altinnPackage = AltinnPackage.from(new ByteArrayInputStream(message));
-            return altinnPackage.getEduDocument();
-
+            DataHandler dh = iBrokerServiceExternalBasicStreamed.downloadFileStreamedBasic(configuration.getUsername(), configuration.getPassword(), request.getFileReference(), request.getReciever());
+            // TODO: rewrite this when Altinn fixes zip
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1135)